cirq/google/calibration/workflow.py

```diff
@@ -269,19 +268,23 @@ def _merge_into_calibrations(
 
 def run_characterizations(
     calibrations: Sequence[PhasedFSimCalibrationRequest],
-    engine: Engine,
-    processor_id: str,
-    gate_set: SerializableGateSet,
+    engine: Union[Engine, PhasedFSimEngineSimulator],
```

Comment:
I really really think this should take a `QuantumEngineSampler` object (and then get rid of `processor_id` and `gate_set`. cc @dstrain115 we may need to add a `run_calibrations` method to q.e.sampler

Reply:
There is a big difference between `cg.Engine.run_calibrations` and this method: `cg.Engine.run_calibrations` works on a proto/message level and this method, as well as `PhasedFSimEngineSimulator`, work on the dataclass/interface level.

If we move `run_calibrations` to q.e.sampler and stick to the proto/message level, than we need to do two more things:
1. Add support for proto-level simulation to the `PhasedFSimEngineSimulator`
2. Somehow hack the XEB implementation so it could work both in the client-side mode as well as the server-side mode.

If we move this method, which works on high-level interface than those problems are gone but we're promoting this functionality to be more primary. I wanted to do it in a second phase of this work after we evaluate its usefulness.

One option would be to keep the `PhasedFSimEngineSimulator` internal to this module for now, and use it only for testing purposes but we're loosing some potential features there.